### DISCUSSION TITLE
Help with creating TermsSets  (Get-PnPTerm/CreateTerm)

### DISCUSSION THREAD

**Discussion by nico-s** [pnp/powershell#851] [Q&A]
Hello, 

I need some help importing our project numbers into the Termstore. 

The hierarchy is Projects/Numbers/Number. The Problem is I can't seem to create a term on the third layer. It will either return a null error or automatically create the term on the fourth layer. 
`
$ParentTerm = Get-PnPTerm -TermGroup "Projects" -TermSet "Numbers" -Identity "Numbers" -IncludeChildTerms -Recursive
$ChildTerm = $ParentTerm.CreateTerm("$Number", 1031, (New-Guid))
Invoke-PnPQuery   
`

How would I create all my terms on the third layer? 

Thanks in advance.
Nico

**Comment by BaronSparky** [comment]
Have you tried using the GUID value for the parent term instead of the displayname/label and then using `Add-PnPTermToTerm`? 

I.e.

`$parentTerm = Get-PnPTerm -TermGroup "Projects" -TermSet "Numbers" -Identity <guid of the parent term> `
`Add-PnPTermToTerm -ParentTermId $parentTerm.Id.Guid -Name "ChildTerm" -Lcid 1031"`

**Reply by nico-s** [reply]
Thanks for the answer. But this results in an error. TermSet and Identity are the same. Is this even possible?
`
$parentTerm = Get-PnPTerm -TermGroup "Projects" -TermSet "Numbers" -Identity "Numbers" (I tried with names and guid)

Add-PnPTermToTerm : Das Argument kann nicht an den Parameter "ParentTermId" gebunden werden, da es NULL ist.
Add-PnPTermToTerm: The argument can't be bound to the parameter "ParentTermId" because its NULL (rough translation)
`

**Reply by BaronSparky** [reply]
Actually, it should be easier than that. You should just need the GUID of the parent term.
![image](https://user-images.githubusercontent.com/11838512/123616634-628ced00-d7fe-11eb-8dce-6ce9f5809aa0.png)
Taking the above example, to add a new term, `"Test4"` under `"Test"`, all I need to do is:
`Add-PnPTermToTerm -ParentTermId b5af214c-5c40-4f38-a947-775365387052 -Name "Test4" -Lcid 1033`
![image](https://user-images.githubusercontent.com/11838512/123617071-cc0cfb80-d7fe-11eb-8de3-182057e86c8f.png)

Perhaps show us your termset? You have a termset called "Numbers" and a term in there called "Numbers"?

(I would also stick with GUIDs as you could have multiple terms in a termset with the same name in different nodes.)

**Comment by nico-s** [comment]
I like to create my Terms directly under App.Nr.. One layer above what you did. 
![terms](https://user-images.githubusercontent.com/8611205/123628205-1dbf8100-d813-11eb-8e40-8d4b98810e65.PNG)

(Yes, I will update my code to what you showed me. )
